### PR TITLE
adapted to new web site layout

### PR DESCRIPTION
**Pull request by fabsenet**
this is a first step to adapting to the new/changed web site layout but my time for today is over.

 - there is a huge issue with date parsing because I change the locale for that and that is discouraged and possibly might break other threads/tasks (i am thinking of home assistant!)
 - there is a local proxy and ssl disabling in the top of the file which needs to be removed before publishing.

but it would maybe able to fix https://github.com/custom-components/sensor.stadtreinigung_hamburg/issues/10 in the end

### CODE REVIEW

File: stadtreinigung_hamburg/StadtreinigungHamburg.py [hunk 1]
```diff
@@ -6,53 +8,85 @@
...
-        uuid = r.html.find("input[name=asId]")[0].attrs["value"]
-        uuid += "-" + r.html.find("input[name=hnId]")[0].attrs["value"]
-
-        for tr in r.html.find("#abfuhrkalender table tr"):
+        uuid = F"{asId}-{hnId}"
+        logger.debug(F"uuid is {uuid}")
+        
+        rows = r.html.find("tbody tr")
+        logger.debug(F"found {len(rows)} garbage collection data rows.")
+        actualLocale = locale.setlocale(locale.LC_TIME, "de_DE")
```
Comment (fabsenet) [inline stadtreinigung_hamburg/StadtreinigungHamburg.py]: That is the critical part, but I have currently no idea to parse "Do 27. Januar 2022" without it. Help?!

**Comment by fabsenet** [comment]
also the first url has a cHash parameter and i have no idea, whether that will change or not

**Comment by vigonotion** [comment]
I tried it successfully without the chash.

The bigger problem is the street asid number hnid conversion. I can't get that request to work. I will push my branch later

**Comment by fabsenet** [comment]
> The bigger problem is the street asid number hnid conversion.

did you noticed that that is working in my code?

**Comment by fabsenet** [comment]
with names:
```shell
❯ python -m stadtreinigung_hamburg "Krähenweg" "41"
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): www.stadtreinigung.hamburg:443
C:\Python310\lib\site-packages\urllib3\connectionpool.py:1013: InsecureRequestWarning: Unverified HTTPS request is being made to host 'localhost'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/1.26.x/advanced-usage.html#ssl-warnings
  warnings.warn(
DEBUG:urllib3.connectionpool:https://www.stadtreinigung.hamburg:443 "POST /abfuhrkalender?tx_srh_pickups%5Baction%5D=addresses&tx_srh_pickups%5Bcontroller%5D=PickUps&type=10002&cHash=549969ec3dc1f708797c7e8762ea3b2b HTTP/1.1" 200 None
DEBUG:stadtreinigung_hamburg.StadtreinigungHamburg:found 1 street(s)
DEBUG:stadtreinigung_hamburg.StadtreinigungHamburg:asId is 988
DEBUG:stadtreinigung_hamburg.StadtreinigungHamburg:there are 26 numbers in this street.
DEBUG:stadtreinigung_hamburg.StadtreinigungHamburg:hnId is 367885
DEBUG:urllib3.connectionpool:Resetting dropped connection: www.stadtreinigung.hamburg
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (2): www.stadtreinigung.hamburg:443
C:\Python310\lib\site-packages\urllib3\connectionpool.py:1013: InsecureRequestWarning: Unverified HTTPS request is being made to host 'localhost'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/1.26.x/advanced-usage.html#ssl-warnings
  warnings.warn(
DEBUG:urllib3.connectionpool:https://www.stadtreinigung.hamburg:443 "GET /abfuhrkalender/?tx_srh_pickups%5Bstreet%5D=988&tx_srh_pickups%5Bhousenumber%5D=367885 HTTP/1.1" 200 125480
DEBUG:stadtreinigung_hamburg.StadtreinigungHamburg:uuid is 988-367885
DEBUG:stadtreinigung_hamburg.StadtreinigungHamburg:found 11 garbage collection data rows.
DEBUG:stadtreinigung_hamburg.StadtreinigungHamburg:['Sa 08. Januar 2022', 'Weihnachtsbäume']
DEBUG:stadtreinigung_hamburg.StadtreinigungHamburg:['Mi 12. Januar 2022', 'Schwarze Restmülltonne']
DEBUG:stadtreinigung_hamburg.StadtreinigungHamburg:['Do 13. Januar 2022', 'Grüne Biotonne']
DEBUG:stadtreinigung_hamburg.StadtreinigungHamburg:['Sa 15. Januar 2022', 'Weihnachtsbäume']
DEBUG:stadtreinigung_hamburg.StadtreinigungHamburg:['Mo 17. Januar 2022', 'Gelbe Wertstofftonne/-sack']
DEBUG:stadtreinigung_hamburg.StadtreinigungHamburg:['Mi 19. Januar 2022', 'Schwarze Restmülltonne']
DEBUG:stadtreinigung_hamburg.StadtreinigungHamburg:['Di 25. Januar 2022', 'Blaue Papiertonne']
DEBUG:stadtreinigung_hamburg.StadtreinigungHamburg:['Mi 26. Januar 2022', 'Schwarze Restmülltonne']
DEBUG:stadtreinigung_hamburg.StadtreinigungHamburg:['Do 27. Januar 2022', 'Grüne Biotonne']
DEBUG:stadtreinigung_hamburg.StadtreinigungHamburg:['Mo 31. Januar 2022', 'Gelbe Wertstofftonne/-sack']
DEBUG:stadtreinigung_hamburg.StadtreinigungHamburg:['Mi 02. Februar 2022', 'Schwarze Restmülltonne']
DEBUG:stadtreinigung_hamburg.StadtreinigungHamburg:succeeded with fetching and parsind!
Next garbage collections:
Weihnachtsbäume on 2022/01/08
Schwarze Restmülltonne on 2022/01/12
Grüne Biotonne on 2022/01/13
Weihnachtsbäume on 2022/01/15
Gelbe Wertstofftonne/-sack on 2022/01/17
Schwarze Restmülltonne on 2022/01/19
Blaue Papiertonne on 2022/01/25
Schwarze Restmülltonne on 2022/01/26
Grüne Biotonne on 2022/01/27
Gelbe Wertstofftonne/-sack on 2022/01/31
Schwarze Restmülltonne on 2022/02/02
```

and with IDs:
```shell
❯ python -m stadtreinigung_hamburg --asid 988  --hnid 367885 
DEBUG:stadtreinigung_hamburg.StadtreinigungHamburg:asId is 988, hnId is 367885
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): www.stadtreinigung.hamburg:443
C:\Python310\lib\site-packages\urllib3\connectionpool.py:1013: InsecureRequestWarning: Unverified HTTPS request is being made to host 'localhost'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/1.26.x/advanced-usage.html#ssl-warnings
  warnings.warn(
DEBUG:urllib3.connectionpool:https://www.stadtreinigung.hamburg:443 "GET /abfuhrkalender/?tx_srh_pickups%5Bstreet%5D=988&tx_srh_pickups%5Bhousenumber%5D=367885 HTTP/1.1" 200 125480
DEBUG:stadtreinigung_hamburg.StadtreinigungHamburg:uuid is 988-367885
DEBUG:stadtreinigung_hamburg.StadtreinigungHamburg:found 11 garbage collection data rows.
DEBUG:stadtreinigung_hamburg.StadtreinigungHamburg:['Sa 08. Januar 2022', 'Weihnachtsbäume']
DEBUG:stadtreinigung_hamburg.StadtreinigungHamburg:['Mi 12. Januar 2022', 'Schwarze Restmülltonne']
DEBUG:stadtreinigung_hamburg.StadtreinigungHamburg:['Do 13. Januar 2022', 'Grüne Biotonne']
DEBUG:stadtreinigung_hamburg.StadtreinigungHamburg:['Sa 15. Januar 2022', 'Weihnachtsbäume']
DEBUG:stadtreinigung_hamburg.StadtreinigungHamburg:['Mo 17. Januar 2022', 'Gelbe Wertstofftonne/-sack']
DEBUG:stadtreinigung_hamburg.StadtreinigungHamburg:['Mi 19. Januar 2022', 'Schwarze Restmülltonne']
DEBUG:stadtreinigung_hamburg.StadtreinigungHamburg:['Di 25. Januar 2022', 'Blaue Papiertonne']
DEBUG:stadtreinigung_hamburg.StadtreinigungHamburg:['Mi 26. Januar 2022', 'Schwarze Restmülltonne']
DEBUG:stadtreinigung_hamburg.StadtreinigungHamburg:['Do 27. Januar 2022', 'Grüne Biotonne']
DEBUG:stadtreinigung_hamburg.StadtreinigungHamburg:['Mo 31. Januar 2022', 'Gelbe Wertstofftonne/-sack']
DEBUG:stadtreinigung_hamburg.StadtreinigungHamburg:['Mi 02. Februar 2022', 'Schwarze Restmülltonne']
DEBUG:stadtreinigung_hamburg.StadtreinigungHamburg:succeeded with fetching and parsind!
Next garbage collections:
Weihnachtsbäume on 2022/01/08
Schwarze Restmülltonne on 2022/01/12
Grüne Biotonne on 2022/01/13
Weihnachtsbäume on 2022/01/15
Gelbe Wertstofftonne/-sack on 2022/01/17
Schwarze Restmülltonne on 2022/01/19
Blaue Papiertonne on 2022/01/25
Schwarze Restmülltonne on 2022/01/26
Grüne Biotonne on 2022/01/27
Gelbe Wertstofftonne/-sack on 2022/01/31
Schwarze Restmülltonne on 2022/02/02
```

File: stadtreinigung_hamburg/StadtreinigungHamburg.py [hunk 2]
```diff
@@ -6,55 +7,97 @@
...
+    text = text.replace("Juni","06.")
+    text = text.replace("Juli","07.")
+    text = text.replace("August","08.")
+    text = text.replace("September","09.")
+    text = text.replace("Oktober","10.")
+    text = text.replace("November","11.")
+    text = text.replace("Dezember","12.")
+
+    return datetime.strptime(text, '%d. %m. %Y')
+
```
Comment (fabsenet) [inline stadtreinigung_hamburg/StadtreinigungHamburg.py:100]: this was the easy way to get rid of locale changing

**Comment by fabsenet** [comment]
i feel like this is ready now for action :-)

wanna push @vigonotion ?

**Comment by vigonotion** [comment]
No I didn't see that! Thanks for your work on that, @fabsenet!

**Pull request merged**